### PR TITLE
Support customising the NATS Streaming channel.

### PR DESCRIPTION
**Pull request by bmcstdio**
## Description

This PR adds support for customising the NATS Streaming channel used for asynchronous invocations, as per openfaas/faas#1399, in a backwards-compatible way.

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))
- [ ] My issue has received approval from the maintainers or lead with the `design/approved` label

https://github.com/openfaas/faas/issues/1399

## How Has This Been Tested?

By running the included unit tests, which is enough in this case IMHO. There's also `bmcstdio/openfaas-gateway:pr-1400`, just in case.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: gateway/Gopkg.toml [hunk 1]
```diff
@@ -12,7 +12,8 @@
 
 [[constraint]]
   name = "github.com/openfaas/nats-queue-worker"
-  version = "0.8.2"
+  source = "https://github.com/bmcstdio/nats-queue-worker"
```
Comment (bmcstdio) [inline gateway/Gopkg.toml]: This will have to be changed once https://github.com/openfaas/nats-queue-worker/pull/79 gets merged.

**Comment by alexellis** [comment]
Please switch over to https://github.com/openfaas/nats-queue-worker/releases/tag/0.9.0 when ready.

**Comment by bmcstdio** [comment]
@alexellis done!

**Comment by alexellis** [comment]
Thank you for the PR

**Comment by alexellis** [comment]
Released in https://github.com/openfaas/faas/releases/tag/0.18.5

**Pull request merged**